### PR TITLE
generate heap profile to data dir

### LINKED ISSUE #11265: Heap profiling result may be lost after pod restarting
## Bug Report

Currently we generate heap profile to `/tmp`, then the files may be lost after pod restarting in docker environment.

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <tangminghua@pingcap.com>

### What problem does this PR solve?

Issue Number: close #11265

Problem Summary:
Currently we generate heap profile to /tmp, then the files may be lost after pod restarting in docker environment.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
* generate heap profile to data dir

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by hicqu — APPROVED** [review]

File: src/server/status_server/mod.rs [hunk 1]
```diff
@@ -116,6 +119,7 @@ where
             router,
             security_config,
             _snap: PhantomData,
+            store_path,
```
Comment (tabokie) [inline src/server/status_server/mod.rs:122]: swap

File: src/server/status_server/mod.rs [hunk 2]
```diff
@@ -673,6 +680,9 @@ where
                             (Method::GET, "/debug/pprof/profile") => {
                                 Self::dump_cpu_prof_to_resp(req).await
                             }
+                            // (Method::GET, "/debug/pprof/heap") => {
+                            //     Self::dump_heap_prof_to_resp(req).await
+                            // }
```
Comment (tabokie) [inline src/server/status_server/mod.rs:685]: remove?

File: src/server/status_server/profile.rs [hunk 3]
```diff
@@ -118,13 +119,20 @@ where
 
 /// Activate heap profile and call `callback` if successfully.
 /// `deactivate_heap_profile` can only be called after it's notified from `callback`.
-pub async fn activate_heap_profile<S, F>(dump_period: S, callback: F) -> Result<(), String>
+pub async fn activate_heap_profile<S, F>(
+    dump_period: S,
+    callback: F,
+    store_path: PathBuf,
```
Comment (tabokie) [inline src/server/status_server/profile.rs:125]: Probably swap too

**Review by tabokie — APPROVED** [review]

File: src/server/status_server/mod.rs (hunk 2 above)
Comment (5kbpers) [reply src/server/status_server/mod.rs]: We are considering to factor it and maybe add it back, so keeping this comment would be better.

**Pull request merged**